R/render.R

```diff
@@ -249,7 +252,20 @@ render <- function(input,
                    envir = parent.frame(),
                    run_pandoc = TRUE,
                    quiet = FALSE,
-                   encoding = "UTF-8") {
+                   encoding = "UTF-8",
+                   customize = FALSE) {
+
+  yaml_knit <- yaml_front_matter(input)[["knit"]]
+  if (!is.null(yaml_knit)) {
+    return(eval(parse(text = yaml_knit))(
+      input = input, output_file = output_file, output_dir = output_dir,
+      output_options = output_options, output_yaml = output_yaml,
+      intermediates_dir = intermediates_dir, knit_root_dir = knit_root_dir,
+      runtime = runtime, clean = clean, knit_meta = knit_meta, envir = envir,
+      run_pandoc = run_pandoc, quiet = quiet, encoding = encoding,
+      customize = FALSE
+    ))
```

Comment:
I think this will create some errors. I don't think we can assume that any function passed through the `knit` yaml option is a function that will take all these arguments ! 

First try was with this small example with new reprex feature
````r
---
output: reprex::reprex_document
knit: reprex::reprex_render
---

Describe you issue very briefly here. Then show it with a minimal, self-contained example in the following R chunk.

```{r}
x <- 1
y <- "2"
x + y
```
````
and I got this error 
```
> render("test.Rmd")
Error in eval(parse(text = yaml_knit))(input = input, output_file = output_file,  : 
  unused arguments (output_file = output_file, output_dir = output_dir, output_options = output_options, output_yaml = output_yaml, intermediates_dir = intermediates_dir, knit_root_dir = knit_root_dir, runtime = runtime, clean = clean, knit_meta = knit_meta, envir = envir, run_pandoc = run_pandoc, quiet = quiet, customize = FALSE)
```` 
as expected. 

I think it will break other use of this knit field when rendering without the knit button with `rmarkdown::render`. like some usage in the wild for [slidify](https://github.com/ramnathv/slidify/commit/07794a3c2d228822c173c1ad5eacf5ab54a42062). I tested and got same type of error as expected. 

Another test is trying to render the bookdown book which have this [knit field](https://github.com/rstudio/bookdown/blob/70f9c07b2049f772b377956047421172069193d9/inst/examples/index.Rmd#L5). I am not able to render the book using `render("index.Rmd")`. I get a weird error about a stack pile. `render` could be called recursively maybe 🤔 
But still it does not work.

Reply:
Thanks for the review!
At first I thought all arguments should be passed, but now am thinking that we only have to pass input.
If then, the code becomes simple, and the behavior becomes much closer to what RStudio's knit button does.
What do you think?

Reply:
Updated via 6c92b6f so that only the input argument is passed.

Reply:
> At first I thought all arguments should be passed, but now am thinking that we only have to pass input.
There need at least to be two fields. That is what the RStudio IDE expect: first argument is the input file name then a encoding field. Like in the [slidify](https://github.com/ramnathv/slidify/commit/07794a3c2d228822c173c1ad5eacf5ab54a42062) example.